src/Symfony/Component/Mailer/Bridge/Mailgun/Mailgun.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mailer\Bridge\Mailgun;
+
+/**
+ * @author Michael Garifullin <garifullin@gmail.com>
+ *
+ * @experimental in 4.3
+ */
+class Mailgun
+{
+    public const REGION_EU = 'EU';
+    public const REGION_US = 'US';
+
+    private const SMTP_DOMAIN_US = 'smtp.mailgun.org';
+    private const SMTP_DOMAIN_EU = 'smtp.eu.mailgun.org';
+
+    private const ENDPOINT_DOMAIN_EU = 'api.eu.mailgun.net';
+    private const ENDPOINT_DOMAIN_US = 'api.mailgun.net';
+
+    private const HTTP_API_ENDPOINT = 'https://%s/v3/%s/messages';
+    private const HTTP_ENDPOINT = 'https://%s/v3/%s/messages.mime';
+
+    /**
+     * @param string $region
+     *
+     * @return string
+     */
+    public static function resolveSmtpDomainByRegion(string $region = self::REGION_US): string
+    {
+        if (self::REGION_EU === $region) {
+            return self::SMTP_DOMAIN_EU;
+        }
+
+        return self::SMTP_DOMAIN_US;
```

Comment:
Should we make sure that the region is always either `EU` or `US`? US might be a reasonable default, but silently falling back to `US` in case we receive garbage does not feel right.

Reply:
just in case to preserve previous behaviour. Before changes it was used US

Reply:
Keeping US as default is no region is set is probably reasonable. But I would expect an error if I set the region to `MiddleEarth`, `Asia`, `Europe` or any string other than `EU` or `US`.

Reply:
is it enough to have just error as undefined index during access mapping const? or you're expecting like native exception?

Reply:
undefined index will only be a notice in production, and won't fail properly. So please add an exception (which also allows providing a better message in dev mode, so better DX).

Reply:
ok I hope RuntimeException will be fine for you